samples/net/secure_mqtt_sensor_actuator/overlay-static-insecure.conf

```diff
@@ -0,0 +1,18 @@
+# Config to disable TLS and DHCPv4, allowing insecure MQTT and a static IP address.
```

Comment:
Could add a simple `overlay-static.conf` as well, so that we can test the sample in more or less default configuration (co communicating securely with public broker), but with static IP addresses. The following config worked out-of-the-box for native_sim or qemu_x86:
```
CONFIG_NET_DHCPV4=n
CONFIG_NET_CONFIG_SETTINGS=y
CONFIG_NET_CONFIG_NEED_IPV4=y
CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
CONFIG_NET_CONFIG_MY_IPV4_GW="192.0.2.2"
CONFIG_DNS_RESOLVER=y
CONFIG_DNS_SERVER_IP_ADDRESSES=y
CONFIG_DNS_SERVER1="192.0.2.2"
```

Reply:
I tested this on hardware, connecting to the public Mosquitto broker and the connection timed out.

Reply:
This is likely because of gateway/dns server addresses, which need to be set accordingly to your network configuration. This example was tailored for the networks we use with native_sim/qemu.

Reply:
DNS server could likely be set to some public one (like `8.8.8.8`), but w/o setting correct gateway address for your local network, you won't be able to reach the internet.

Reply:
Thanks @rlubos 👍